internal/expr/parser.go

```diff
@@ -386,6 +392,44 @@ func (p *Parser) parseColumn() (fullName, bool, error) {
 	return fullName{}, false, nil
 }
 
+func (p *Parser) parseFuncName() (string, bool, error) {
+	cp := p.save()
+	if p.skipName() {
+		if p.skipByte('(') {
+			bracketLevel := 1
+			for {
```

Comment:
Just to make the breaking conditions a little clearer, could a condition be used instead? Like
```go
for p.pos != len(p.input) && bracketLevel != 0 {
```

Reply:
I've actually just changed it to:
```go
if p.pos == len(p.input) {
	return "", false, nil
}
```
In this case we probably want to actually return false (an error is probably a bad idea since the thing we are parsing could well not be a function).